File: frontend/src/types/job.ts
export type JobStatus =
  | 'uploaded'
  | 'queued'
  | 'running'
  | 'succeeded'
  | 'failed'
  | 'cancelled'
  | (string & {})

export type Job = {
  id: string
  original_filename: string
  status: JobStatus
  progress: number
  stage: string | null
  message: string | null
  error: string | null
  input_video_key: string
  output_csv_key: string | null
  preview_json_key: string | null
  crop_keys_json: string[] | null
  stats_json: Record<string, unknown> | null
  pipeline_name: string
  pipeline_version: string
  created_at: string
  updated_at: string
}

export type JobCrop = {
  key: string
  url: string
  filename: string
}

export type PreviewPayload = unknown

File: backend/app/schemas/job.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class JobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    original_filename: str
    status: str
    progress: int
    stage: str | None
    message: str | None
    error: str | None
    input_video_key: str
    output_csv_key: str | None
    preview_json_key: str | None
    crop_keys_json: list[str] | None
    stats_json: dict[str, Any] | None
    pipeline_name: str
    pipeline_version: str
    created_at: datetime
    updated_at: datetime


class JobCropRead(BaseModel):
    key: str
    url: str
    filename: str
